Src/Plugins/Security/PostgreSqlConnectionStringValidator.cs

```diff
@@ -0,0 +1,92 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Data.Common;
+using System.Text;
+
+using Microsoft.RE2.Managed;
+
+using Npgsql;
+
+namespace Microsoft.CodeAnalysis.Sarif.PatternMatcher.Plugins.Security
+{
+    public class PostgreSqlConnectionStringValidator : ValidatorBase
+    {
+        internal static PostgreSqlConnectionStringValidator Instance;
+        internal static IRegex RegexEngine;
+
+        static PostgreSqlConnectionStringValidator()
+        {
+            Instance = new PostgreSqlConnectionStringValidator();
+            RegexEngine = RE2Regex.Instance;
+        }
+
+        public static string IsValidStatic(ref string matchedPattern,
+                                           ref Dictionary<string, string> groups,
+                                           ref string failureLevel,
+                                           ref string fingerprint,
+                                           ref string message)
+        {
+            return ValidatorBase.IsValidStatic(Instance,
+                                               ref matchedPattern,
+                                               ref groups,
+                                               ref failureLevel,
+                                               ref fingerprint,
+                                               ref message);
+        }
+
+        public static string IsValidDynamic(ref string fingerprint, ref string message)
+        {
+            return ValidatorBase.IsValidDynamic(Instance,
+                                                ref fingerprint,
+                                                ref message);
+        }
+
+        protected override string IsValidStaticHelper(ref string matchedPattern, ref Dictionary<string, string> groups, ref string failureLevel, ref string fingerprintText, ref string message)
+        {
+            if (!groups.TryGetValue("host", out string host) ||
+                !groups.TryGetValue("database", out string database) ||
+                !groups.TryGetValue("account", out string account) ||
+                !groups.TryGetValue("password", out string password))
+            {
+                return nameof(ValidationState.NoMatch);
+            }
+
+            fingerprintText = new Fingerprint()
+            {
+                Host = host.Replace("\"", string.Empty).Replace(",", ";"),
```

Comment:
>.Replace("\"", string.Empty).Replace(",", ";") [](start = 27, length = 46)

we can remove this as well

Reply:
Removed

---
In reply to: [568863442](https://github.com/microsoft/sarif-pattern-matcher/pull/157#discussion_r568863442) [](ancestors = 568863442)